staging/src/k8s.io/client-go/plugin/pkg/client/auth/azure/azure.go

```diff
@@ -210,33 +211,62 @@ func (ts *azureTokenSource) Token() (*azureToken, error) {
 
 	var err error
 	token := ts.cache.getToken(azureTokenKey)
```

Comment:
should we return early if we got a good token from the cache? the code currently always writes non-expired tokens back to the cache

```go
if token != nil && !token.token.IsExpired() {
  return token, nil
}
```